### PR TITLE
FEATURE: SD-909 Add transparency to pushpin and attribution to map

### PR DESCRIPTION
**Pull request by naibinh**
Description of changes (if multi-commit, short global summary & context;
if single-commit, feel free to leave empty).
- Add 70% transparency to pins
- Add attribute to map
- Restructure storybook cases of Points and Viewport
- Update storybook images to new ones with above changes

---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [x] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [x] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [x] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [x] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer:
   - https://github.com/gooddata/gdc-analytical-designer/pull/2999
   - https://client-demo.na.intgdc.com:50029/analyze/
- gdc-dashboards:
   - https://github.com/gooddata/gdc-dashboards/pull/2716
   - https://client-demo.na.intgdc.com:50030/dashboards/

# Related Jira tasks

### CODE REVIEW

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6942/ : FAILURE in 6m 31s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2842/ : SUCCESS in 3s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3626/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2843/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2993/ : UNSTABLE in 19m 01s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6943/ : SUCCESS in 6m 05s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2844/ : SUCCESS in 4s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3627/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2845/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2994/ : UNSTABLE in 18m 35s

**Comment by naibinh** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1530.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build in pipeline check-extended aborted.

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6945/ : SUCCESS in 12m 56s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2849/ : SUCCESS in 7s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3629/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2850/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2997/ : UNSTABLE in 15m 40s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6946/ : SUCCESS in 8m 45s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2851/ : SUCCESS in 4s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3630/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2852/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2998/ : SUCCESS in 19m 10s

**Comment by naibinh** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1530.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2853/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/2999/ : ABORTED in 1h 00m 28s

**Comment by naibinh** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1530.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

File: .storybook/test-storybook-canvas.config.js [hunk 1]
```diff
@@ -61,6 +67,20 @@ const configuration = {
             }),
             {},
         ),
+        "Core components/GeoPushpinChart/Config/Map": mapStories.reduce(
```
Comment (ptquang86) [inline .storybook/test-storybook-canvas.config.js:70]: should it be 
`"Core components/GeoPushpinChart/Config/Viewport": viewportStories.reduce(`? 
We don’t have `map` property in `IGeoConfig`

File: src/components/core/geoChart/tests/geoChartColor.spec.ts [hunk 2]
```diff
@@ -162,11 +162,11 @@ describe("getPushpinColors", () => {
         });
         const colorStrategy: IColorStrategy = createColorStrategy(executionResponse, geoConfig, segmentItems);
         const expectedColors: IPushpinColor[] = [
-            { background: "rgb(215,242,250)", border: "rgb(20,178,226)" },
-            { background: "rgb(212,244,236)", border: "rgb(0,193,141)" },
-            { background: "rgb(237,136,129)", border: "rgb(229,77,66)" },
-            { background: "rgb(241,134,0)", border: "rgb(241,134,0)" },
-            { background: "rgb(171,85,163)", border: "rgb(171,85,163)" },
+            { background: "rgba(215,242,250,0.7)", border: "rgba(20,178,226,0.7)" },
```
Comment (ptquang86) [inline src/components/core/geoChart/tests/geoChartColor.spec.ts:165]: please double check the color of `border`.
We don't need transparency for it :)

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2870/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3006/ : SUCCESS in 20m 15s

File: src/components/visualizations/utils/color.ts [hunk 3]
```diff
@@ -160,7 +161,10 @@ function lighter(color: number, percent: number) {
     return Math.round((t - color) * p) + color;
 }
 
-function formatColor(red: number, green: number, blue: number) {
+function formatColor(red: number, green: number, blue: number, opacity?: number): string {
```
Comment (ptquang86) [inline src/components/visualizations/utils/color.ts:164]: `formatColor` should be consistent with `getCalculatedColors` and `getColorPalette`
Please add the default value for `opacity`

File: src/components/visualizations/utils/color.ts [hunk 4]
```diff
@@ -300,13 +312,18 @@ function getRGBColorCode(color: string | IColor): IColor {
...
     const { r, g, b } = getRGBColorCode(baseColor);
     const channels = [r, g, b];
     const steps = channels.map(channel => (255 - channel) / colorItemsCount);
-    const generatedColors = getCalculatedColors(colorItemsCount, channels, steps);
-    return [...generatedColors.reverse(), formatColor(r, g, b)];
+    const generatedColors = getCalculatedColors(colorItemsCount, channels, steps, opacity);
+    return [...generatedColors.reverse(), formatColor(r, g, b, opacity)];
+}
+
+export function rgbToRgba(color: string, opacity: number = 1) {
```
Comment (ptquang86) [inline src/components/visualizations/utils/color.ts:324]: please add the return type for this function :)

File: tsconfig.json [hunk 5]
```diff
@@ -12,9 +12,9 @@
     ],
     "module": "commonjs",
     "moduleResolution": "node",
-    "noImplicitAny": true,
-    "noUnusedLocals": true,
-    "noUnusedParameters": true,
+//    "noImplicitAny": true,
+//    "noUnusedLocals": true,
+//    "noUnusedParameters": true,
```
Comment (ptquang86) [inline tsconfig.json:17]: Do you want to remove this?

**Comment by naibinh** [comment]
extended test - storybook

File: tsconfig.json (hunk 5 above)
Comment (naibinh) [reply tsconfig.json]: My bad.
Updated

File: src/components/visualizations/utils/color.ts (hunk 4 above)
Comment (naibinh) [reply src/components/visualizations/utils/color.ts]: Updated

File: src/components/visualizations/utils/color.ts (hunk 3 above)
Comment (naibinh) [reply src/components/visualizations/utils/color.ts]: Updated

File: .storybook/test-storybook-canvas.config.js (hunk 1 above)
Comment (naibinh) [reply .storybook/test-storybook-canvas.config.js]: Updated.
I followed by Map under Configuration Panel.

File: src/components/core/geoChart/tests/geoChartColor.spec.ts (hunk 2 above)
Comment (naibinh) [reply src/components/core/geoChart/tests/geoChartColor.spec.ts:165]: As we also update the color palette and border take the darkest one in it, so border gets affected as well.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6964/ : SUCCESS in 7m 04s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2880/ : SUCCESS in 3s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3639/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2881/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3011/ : SUCCESS in 19m 21s

**Comment by naibinh** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6974/ : SUCCESS in 8m 56s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2895/ : SUCCESS in 3s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3647/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2898/ : SUCCESS in 5s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3019/ : SUCCESS in 24m 34s

**Comment by ptquang86** [comment]
LGTM

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6981/ : SUCCESS in 6m 22s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2906/ : SUCCESS in 3s

**Comment by dlcd2203** [comment]
LGTM

**Comment by naibinh** [comment]
extended test - storybook

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3654/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/2916/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3026/ : SUCCESS in 18m 51s

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3656/ : SUCCESS in 15m 26s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/6992/ : SUCCESS in 19m 26s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1692/ : SUCCESS in 40s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2459/ : SUCCESS in 24m 25s

**Pull request merged**